src/modules/editor/sagas.ts

```diff
@@ -303,49 +328,71 @@ function* handleOpenEditor(action: OpenEditorAction) {
   // The client will report when an entity goes out of bounds
   yield call(() => editorWindow.editor.on('entitiesOutOfBoundaries', handleEntitiesOutOfBoundaries))
 
-  // Creates a new scene in the dcl client's side
-  const project: Project | Pool | null = yield type === 'pool' ? select(getCurrentPool) : select(getCurrentProject)
-
-  if (project) {
-    // load asset packs
-    const areLoaded = yield select(hasLoadedAssetPacks)
-    if (!areLoaded) {
-      yield put(loadAssetPacksRequest())
+  if (type === PreviewType.WEARABLE) {
+    const search = yield select(getSearch)
+    const itemId = new URLSearchParams(search).get('item')
+    const items: Item[] = yield select(getItems)
+    const item = items.find(item => item.id === itemId)
+    if (item) {
+      yield put(setEditorReadOnly(true))
+      yield createNewEditorScene(createAvatarProject())
+
+      // set camera
+      yield call(async () => {
+        editorWindow.editor.resetCameraZoom()
+        editorWindow.editor.setCameraPosition({ x: 8, y: 1.2, z: 8 })
+        editorWindow.editor.setCameraZoomDelta(-30)
+        editorWindow.editor.setCameraRotation(Math.PI, Math.PI / 16)
+      })
+
+      // render selected items
+      yield put(setItems([item]))
     }
+  } else {
+    // Creates a new scene in the dcl client's side
+    const project: Project | Pool | null = yield type === PreviewType.POOL ? select(getCurrentPool) : select(getCurrentProject)
+
+    if (project) {
+      // load asset packs
+      const areLoaded = yield select(hasLoadedAssetPacks)
+      if (!areLoaded) {
+        yield put(loadAssetPacksRequest())
+      }
 
-    // fix legacy stuff
-    let scene: Scene = yield getSceneByProjectId(project.id, type)
-    yield put(fixLegacyNamespacesRequest(scene))
-    const fixSuccessAction: FixLegacyNamespacesSuccessAction = yield take(FIX_LEGACY_NAMESPACES_SUCCESS)
-    scene = fixSuccessAction.payload.scene
+      // fix legacy stuff
+      let scene: Scene = yield getSceneByProjectId(project.id, type)
+      yield put(fixLegacyNamespacesRequest(scene))
+      const fixSuccessAction: FixLegacyNamespacesSuccessAction = yield take(FIX_LEGACY_NAMESPACES_SUCCESS)
+      scene = fixSuccessAction.payload.scene
 
-    // if assets packs are being loaded wait for them to finish
-    const state: RootState = yield select(state => state)
-    if (isLoadingType(state.assetPack.loading, LOAD_ASSET_PACKS_REQUEST)) {
-      yield take(LOAD_ASSET_PACKS_SUCCESS)
-    }
+      // if assets packs are being loaded wait for them to finish
+      const state: RootState = yield select(state => state)
```

Comment:
Why not just getting the assetPack loading state?

Reply:
fixed 👍